ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartCompleteUploadHandler.java

```diff
@@ -382,12 +383,27 @@ private boolean isRetriable(Throwable throwable) {
     List<ChunkInfo> getChunksToStitch(CompleteMultipartUpload completeMultipartUpload) throws RestServiceException {
       // Get parts in order from CompleteMultipartUpload, deserialize each part id to get data chunk ids.
       List<ChunkInfo> chunkInfos = new ArrayList<>();
+      Set<String> eTags = new HashSet<>();
       try {
         // sort the list in order
         List<Part> sortedParts = Arrays.asList(completeMultipartUpload.getPart());
         Collections.sort(sortedParts, Comparator.comparingInt(Part::getPartNumber));
         String reservedMetadataId = null;
         for (Part part : sortedParts) {
+          // Validate part number: must be a non-negative integer and within the range 0-10,000
```

Comment:
put a link to s3 doc
https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html